presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcRecordCursorFactory.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc;
+
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.RecordCursor;
+
+import java.util.List;
+
+public interface JdbcRecordCursorFactory
```

Comment:
We have JdbcRecordSetProvider, JdbcRecordSet and JdbcRecordCursorFactory, each being basically the same thing.

Instead of providing the new binding, you could make JdbcRecordSetProvider configurable.

Would that work?

Reply:
> Instead of providing the new binding, you could make JdbcRecordSetProvider configurable.

Do you mean that I should replace `JdbcRecordCursorFactory` with `JdbcRecordSetFactory` so I could have an option to customize `JdbcRecordSetProvider#getRecordSet`? Surely that would work.